src/js/common/model/veteran.js

```diff
@@ -717,6 +717,10 @@ function veteranToApplication(veteran) {
     /* eslint-enable no-param-reassign */
   }
 
+  if (veteran.discloseFinancialInformation.value === 'N') {
+    delete veteran.children; // eslint-disable-line no-param-reassign
```

Comment:
Can we set this to an empty array instead? That's how it's done in the conformance tests we have in the backend, and even though they work with the element missing I'd prefer to not chance anything..